File: hospital-appointment-frontend/src/api/hospital/review.ts
/**
 * 就医评价 API
 */

import request from '../request'

export interface Review {
  reviewID: string
  patientID: number
  patientName?: string
  doctorID: number
  doctorName?: string
  appointmentID: string
  departmentID: number
  departmentName?: string
  visitID?: number
  overallRating: number
  attitudeRating?: number
  skillRating?: number
  environmentRating?: number
  reviewContent?: string
  reviewImages?: string
  replyContent?: string
  replyTime?: string
  isAnonymous: number
  reviewStatus: number
  createTime: string
}

export interface CreateReviewData {
  patientID: number
  doctorID: number
  appointmentID: string
  departmentID: number
  visitID?: number
  overallRating: number
  attitudeRating?: number
  skillRating?: number
  environmentRating?: number
  reviewContent?: string
  reviewImages?: string
  isAnonymous?: number
}

export const reviewAPI = {
  /**
   * 提交评价
   */
  submit: (data: CreateReviewData) => {
    return request.post<string>('/hospital/review', data)
  },

  /**
   * 获取医生的评价列表
   */
  getByDoctor: (doctorId: number, page: number = 1, size: number = 10) => {
    return request.get<Review[]>(`/hospital/review/doctor/${doctorId}`, {
      params: { page, size }
    })
  },

  /**
   * 获取患者的历史评价
   */
  getByPatient: (patientId: number) => {
    return request.get<Review[]>(`/hospital/review/patient/${patientId}`)
  },

  /**
   * 获取评价详情
   */
  getById: (id: string) => {
    return request.get<Review>(`/hospital/review/${id}`)
  },

  /**
   * 获取医生的平均评分
   */
  getAverageRating: (doctorId: number) => {
    return request.get<number>(`/hospital/review/doctor/${doctorId}/rating`)
  },

  /**
   * 医生回复评价
   */
  reply: (id: string, reply: string) => {
    return request.put<string>(`/hospital/review/${id}/reply`, null, {
      params: { reply }
    })
  },

  /**
   * 审核评价（管理员）
   */
  audit: (id: string, status: number) => {
    return request.put<string>(`/hospital/review/${id}/audit`, null, { params: { status } })
  },

  /**
   * 获取待审核评价列表（管理员）
   */
  getPending: () => {
    return request.get<Review[]>('/hospital/review/pending')
  }
}

File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/Review.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 就医评价实体类
 * 改造自反馈表(feedback)
 */
@Data
@TableName("review")
public class Review implements Serializable {
    
    @TableId(type = IdType.INPUT)
    private String reviewID;
    
    /** 患者ID */
    private Long patientID;
    
    /** 医生ID */
    private Long doctorID;
    
    /** 挂号记录ID */
    private String appointmentID;
    
    /** 科室ID */
    private Long departmentID;
    
    /** 就诊记录ID */
    private Long visitID;
    
    /** 总体评分 1-5 */
    private Integer overallRating;
    
    /** 服务态度评分 1-5 */
    private Integer attitudeRating;
    
    /** 医疗技术评分 1-5 */
    private Integer skillRating;
    
    /** 就医环境评分 1-5 */
    private Integer environmentRating;
    
    /** 评价内容 */
    private String reviewContent;
    
    /** 评价图片 */
    private String reviewImages;
    
    /** 医生回复 */
    private String replyContent;
    
    /** 回复时间 */
    private LocalDateTime replyTime;
    
    /** 是否匿名：0-否 1-是 */
    private Integer isAnonymous;
    
    /** 状态：1-待审核 2-已发布 3-已隐藏 */
    private Integer reviewStatus;
    
    /** 评价时间 */
    private LocalDateTime createTime;
    
    /** 非数据库字段：医生姓名 */
    @TableField(exist = false)
    private String doctorName;
    
    /** 非数据库字段：科室名称 */
    @TableField(exist = false)
    private String departmentName;
    
    /** 非数据库字段：患者姓名 */
    @TableField(exist = false)
    private String patientName;
}
